### PR TITLE
Image show page

### PR DESCRIPTION
**Pull request by conproctor**

### CODE REVIEW

**Comment by rmacklin** [comment]
The `width: 100%` style can make some of the smaller images look pixelated when they're stretched to full width on the show page. You don't have to change this if you don't mind the look, I just wanted to mention it in case you'd prefer to constrain the width.

**Comment by conproctor** [comment]
I was going to change it but then I saw one of the puppies blown up massive and it was super cute, so I left it.

**Comment by rmacklin** [comment]
Haha sounds good. My other comment is that I think it would be better to not require a hover to see the tags and delete button when you're viewing the show page. One way that you could do this is to use a custom class instead of `card-img-overlay`, and `@extend` the `.card-img-overlay` styles when the card is rendered in the index page context.

File: test/helpers/application_helper_test.rb [hunk 1]
```diff
@@ -16,8 +16,7 @@ class ApplicationHelperTest < ActionView::TestCase
     image = Image.create!(url: url, tag_list: 'cute')
     card = image_card(image)
     expected_card = <<-HTML.sub(/\n$/, '')
-<div class="card" data-image-id="1"><img class="card__image img-fluid" src="#{url}" alt="Puppy 500x350" />
-<div class="card-img-overlay">
+<div class="card" data-image-id="1"><a href="/images/1"><img class="card__image img-fluid" src="http://media.mydogspace.com.s3.amazonaws.com/wp-content/uploads/2013/08/puppy-500x350.jpg" alt="Puppy 500x350" /></a><div class="card-img-overlay">
```
Comment (rmacklin) [inline test/helpers/application_helper_test.rb]: I think you're getting lucky here that the `image` created always has id `1`. It would be better to interpolate `image.id` everywhere you've got the image id hardcoded to 1.

Comment (conproctor) [inline test/helpers/application_helper_test.rb]: 👍

**Comment by conproctor** [comment]
Okay I made it so they are always visible when on the show page: c54a537

File: app/assets/stylesheets/_card.scss [hunk 2]
```diff
@@ -9,6 +9,11 @@
   pointer-events: none;
 }
 
+.card-img-overlay__no_hover {
+  @extend .card-img-overlay;
+  display: block;
+}
+
```
Comment (rmacklin) [inline app/assets/stylesheets/_card.scss]: Instead of this, consider using a state class on the `.card` element. You would modify line 17 to:
```css
.card:hover .card-img-overlay, .card.is-showing-overlay .card-img-overlay {
```

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -7,9 +7,9 @@ def nav_item(text:, link:, is_active: false)
     end
   end
 
-  def image_card(image)
+  def image_card(image, hover_for_overlay: true)
     content_tag :div, class: 'card', 'data-image-id': image.id do
-      render partial: 'images/card_contents', locals: { image: image }
+      render partial: 'images/card_contents', locals: { image: image, hover_for_overlay: hover_for_overlay }
```
Comment (rmacklin) [inline app/helpers/application_helper.rb]: Following the previous comment (https://github.com/AppFolioOnboarding/image-sharer-conproctor/pull/30#discussion_r61131821), you'd undo this change and instead have:
```ruby
classes = ['card']
classes << 'is-showing-overlay' unless hover_for_overlay
content_tag :div, class: classes, 'data-image-id': image.id do
```

You could also change the argument to `show_overlay: true` flip the condition to:
```ruby
classes << 'is-showing-overlay' if show_overlay
```

File: app/views/images/_card_contents.html.erb [hunk 4]
```diff
@@ -1,7 +1,11 @@
 <%= link_to image_path(image) do %>
   <% image_tag image.url, class: 'card__image img-fluid' %>
 <% end %>
-<div class="card-img-overlay">
+<% if hover_for_overlay %>
+  <div class="card-img-overlay">
+<% else %>
+  <div class="card-img-overlay__no_hover">
+<% end %>
```
Comment (rmacklin) [inline app/views/images/_card_contents.html.erb]: Following the previous comments (https://github.com/AppFolioOnboarding/image-sharer-conproctor/pull/30#discussion_r61131821 and https://github.com/AppFolioOnboarding/image-sharer-conproctor/pull/30#discussion_r61132116) this would also get reverted.
